### ISSUE TITLE
train on single dataset?

### ISSUE THREAD

**Issue by zhanganny** [g-fiche/VQ-HPS#1]
Thanks for your great work! How to train on e 3DPW, COCO, and EMDB training sets separately? Could you release the details about "training on limited data"?

**Comment by g-fiche** [comment]
Thank you for your interest in our work!

To train on a single dataset, you will need to modify the [config file](https://github.com/g-fiche/VQ-HPS/blob/master/configs/config.yaml): the training data is specified L35.

For training on COCO, we already provide "coco_train.txt"; you can create files for other datasets using the same format.

Regarding the experiment on EMDB, we split EMDB2 into a training and a validation set. I will add the code for doing that in the [preprocessing file](https://github.com/g-fiche/VQ-HPS/blob/master/preprocess_data/emdb.py).

I hope this answers to your questions, please do not hesitate to ask for more information if needed!

**Comment by zhanganny** [comment]
Thank you for your quick reply! How about training on the 3DPW dataset? Could you please also add the code for preprocessing it?

**Comment by zhanganny** [comment]
> Thank you for your interest in our work!
> 
> To train on a single dataset, you will need to modify the [config file](https://github.com/g-fiche/VQ-HPS/blob/master/configs/config.yaml): the training data is specified L35.
> 
> For training on COCO, we already provide "coco_train.txt"; you can create files for other datasets using the same format.
> 
> Regarding the experiment on EMDB, we split EMDB2 into a training and a validation set. I will add the code for doing that in the [preprocessing file](https://github.com/g-fiche/VQ-HPS/blob/master/preprocess_data/emdb.py).
> 
> I hope this answers to your questions, please do not hesitate to ask for more information if needed!

Thank you for your quick reply! How about training on the 3DPW dataset? Is it correct to directly set the config file with training_data:
  file: '3DPW/3DPW_train.npz'?

**Comment by g-fiche** [comment]
The preprocessing code for 3DPW can be found [here](https://github.com/g-fiche/VQ-HPS/blob/master/preprocess_data/threedpw.py). You can run it as follows:
`python preprocess_data/threedpw.py`

This will create the training, validation, and test annotation files in ```datasets/3DPW```.

Then, you have 2 options:

1. Create a ```3dpw_train.txt``` file, which would be a copy of ```coco_train.txt``` except that the path would be ```datasets/3DPW/3DPW_train.npz```. Then, replace the training file in the config file as explained in my first response.
2. Another option is to put directly ```datasets/3DPW/3DPW_train.npz``` as the path in the config file. Then, you will need to modify the [training script](https://github.com/g-fiche/VQ-HPS/blob/master/train_mesh_regressor.py) taking inspiration from the validation dataset (but with augmentations, and ```proportion=1```).

If you choose the second option, the code will look like that:
```
training_data = DatasetHMR(
        dataset_file=f"{path}/{cfg.training_data.file}",
        augment=True,
        flip=True,
        proportion=1,
    )

validation_data = DatasetHMR(
        dataset_file=f"{path}/{cfg.validation_data.file}",
        augment=False,
        proportion=0.1,
    )
```

**Comment by zhanganny** [comment]
> * `3dpw_train.txt` file, which would be a copy of `coco_train.txt` except that the path would be `datasets/3DPW/3DPW_train.npz`. Then, replace the training file in the config file as explained in my first response.
> * Another option is to put directly `datasets/3DPW/3DPW_train.npz` as the path in the config file. Then, you will need to modify the [training script](https://github.com/g-fiche/VQ-HPS/blob/master/train_mesh_regressor.py) taking inspiration fr

OK! I have got it! Thank you so much!
